### PR TITLE
Add e2e test for sampling storage

### PR DESCRIPTION
**Pull request by slayer321**
## Which problem is this PR solving?
- Related #3305

## Description of the changes
-  Adding e2e test for Implemented Storage backends for adaptive sampling i.e (cassendra, in-memory)

## How was this change tested?
- 

## Checklist
- [x] I have read https://github.com/jaegertracing/jaeger/blob/master/CONTRIBUTING_GUIDELINES.md
- [x] I have signed all commits
- [x] I have added unit tests for the new functionality
- [x] I have run lint and test steps successfully
  - for `jaeger`: `make lint test`
  - for `jaeger-ui`: `yarn lint` and `yarn test`

### CODE REVIEW

File: plugin/storage/integration/integration.go [hunk 1]
```diff
@@ -393,6 +396,59 @@ func (s *StorageIntegration) testGetDependencies(t *testing.T) {
...
+		actual, err := s.SamplingStore.GetThroughput(start, start.Add(time.Second*time.Duration(10)))
+		require.NoError(t, err)
+		fmt.Printf("actual: %v\n", actual)
+		fmt.Printf("expected: %v\n", expected)
+		return assert.ObjectsAreEqualValues(expected, len(actual))
+	})
+	if !assert.True(t, found) {
+		t.Log("\t Expected:", expected)
+		t.Log("\t Actual  :", actual)
+	}
```
Comment (yurishkuro) [inline plugin/storage/integration/integration.go:420]:
```suggestion
	assert.Len(t, actual, expected)
```

File: plugin/storage/integration/cassandra_test.go [hunk 2]
```diff
@@ -134,4 +138,5 @@ func TestCassandraStorage(t *testing.T) {
 	require.NoError(t, s1.initializeCassandra())
 	// TODO: Support all other tests.
 	t.Run("GetDependencies", s1.testCassandraGetDependencies)
+	s1.IntegrationTestAll(t)
```
Comment (yurishkuro) [inline plugin/storage/integration/cassandra_test.go:141]: oh, wow, we never tested Cassandra?

Comment (slayer321) [reply plugin/storage/integration/cassandra_test.go:141]: yes,`IntegrationTestAll` was never called for cassendra

File: plugin/storage/integration/integration.go [hunk 3]
```diff
@@ -422,4 +477,10 @@ func (s *StorageIntegration) IntegrationTestAll(t *testing.T) {
 	t.Run("GetLargeSpans", s.testGetLargeSpan)
 	t.Run("FindTraces", s.testFindTraces)
 	t.Run("GetDependencies", s.testGetDependencies)
+	// Remove the below line check once we have sampling store implemented for all the backend
+	if os.Getenv("STORAGE") != "cassendra" || os.Getenv("STORAGE") != "in-memory" {
```
Comment (yurishkuro) [inline plugin/storage/integration/integration.go:481]: this is not how we exclude tests. When a backend does not support sampling, it cannot populate `s.SamplingStore`, so the test can simply check for this being nil and call Skip with appropriate message.

File: plugin/storage/integration/integration.go [hunk 4]
```diff
@@ -126,7 +130,8 @@ func (s *StorageIntegration) testGetServices(t *testing.T) {
 
 	var actual []string
 	found := s.waitForCondition(t, func(t *testing.T) bool {
-		actual, err := s.SpanReader.GetServices(context.Background())
+		var err error
+		actual, err = s.SpanReader.GetServices(context.Background())
```
Comment (yurishkuro) [inline plugin/storage/integration/integration.go:134]: unnecessary change

File: plugin/storage/integration/memstore_test.go [hunk 5]
```diff
@@ -36,8 +37,10 @@ func (s *MemStorageIntegrationTestSuite) initialize() error {
 	s.logger, _ = testutils.NewLogger()
 
 	store := memory.NewStore()
+	sampleStore := memory.NewSamplingStore(2)
 	s.SpanReader = store
 	s.SpanWriter = store
+	s.SamplingStore = sampleStore
```
Comment (yurishkuro) [inline plugin/storage/integration/memstore_test.go:43]: can combine L40 and 43 in one statement

File: plugin/storage/integration/memstore_test.go [hunk 6]
```diff
@@ -57,5 +60,10 @@ func (s *MemStorageIntegrationTestSuite) cleanUp() error {
 func TestMemoryStorage(t *testing.T) {
 	s := &MemStorageIntegrationTestSuite{}
 	require.NoError(t, s.initialize())
+	// Remove the below line check once we have sampling store implemented for all the storage backend
+	err := os.Setenv("STORAGE", "in-memory")
+	if err != nil {
+		t.Fatal(err)
+	}
```
Comment (yurishkuro) [inline plugin/storage/integration/memstore_test.go:67]: not needed

File: plugin/storage/integration/integration.go (hunk 4 above)
Comment (slayer321) [reply plugin/storage/integration/integration.go:134]: adding the new variable named `err` is required because  if we don't do that then a new `actual` variable get's Initialized at L134 and if the `GetServices` method fails we get the empty slice for the actual value as you can see below.

```
    integration.go:139: 
        	Error Trace:	/home/infracloud/go/src/github.com/jaeger/plugin/storage/integration/integration.go:139
        	Error:      	Should be true
        	Test:       	TestMemoryStorage/GetServices
    integration.go:140: 	 Expected: [example-service-1 example-service-2 example-service-3123]
    integration.go:141: 	 Actual  : []

```

File: plugin/storage/integration/integration.go [hunk 7]
```diff
@@ -422,4 +476,9 @@ func (s *StorageIntegration) IntegrationTestAll(t *testing.T) {
 	t.Run("GetLargeSpans", s.testGetLargeSpan)
 	t.Run("FindTraces", s.testFindTraces)
 	t.Run("GetDependencies", s.testGetDependencies)
+	if s.SamplingStore == nil {
```
Comment (yurishkuro) [inline plugin/storage/integration/integration.go:479]: this should be done inside the individual tests that depend on sampling store, not in the top runner

Comment (slayer321) [reply plugin/storage/integration/integration.go:479]: I also thought of that but then I realized that in all the individual tests we are calling `s.IntegrationTestAll(t)` which will run the integration test for them and we can skip sampling store test at the top runner level. 
If I try to skip it at the individual level I'm not sure how can we skip this two specific sampling store test because there we can check for the samplingStore is nil or not .. but we cannot specify that we want to skip this two test.

File: plugin/storage/integration/integration.go [hunk 8]
```diff
@@ -414,6 +420,56 @@ func (s *StorageIntegration) testGetDependencies(t *testing.T) {
 	assert.EqualValues(t, expected, actual)
 }
 
+// === Sampling Store Integration Tests ===
+
+func (s *StorageIntegration) testGetThroughput(t *testing.T) {
+	s.skipIfNeeded(t)
```
Comment (yurishkuro) [inline plugin/storage/integration/integration.go:426]:
```suggestion
	s.skipIfNeeded(t)
	if s.SamplingStore == nil {
	   // skip
	}
```
